pkg/operatorsource/registryhelper.go

```diff
@@ -0,0 +1,37 @@
+package operatorsource
+
+import (
+	"context"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/appregistry"
+
+	corev1 "k8s.io/api/core/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SetupRegistryOptions generates an Options object based on the OperatorSource spec. It passes along
+// the opsrc endpoint and, if defined, retrieves the authorization token from the specified Secret
+// object.
+func SetupRegistryOptions(client k8sclient.Client, spec *v1alpha1.OperatorSourceSpec) (appregistry.Options, error) {
```

Comment:
Please call the file `appregistryhelper.go` and the function `SetupAppRegistryOptions`

Reply:
updated.